File: Models/Experience.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace JobFairPortal.Models
{
    public class Experience
    {
        [Key]
        public int ExperienceId { get; set; }

        [Required]
        [ForeignKey("Student")]
        public int StudentId { get; set; }

        [Required]
        [MaxLength(100)]
        public string CompanyName { get; set; } = null!;

        [Required]
        [MaxLength(100)]
        public string Role { get; set; } = null!;

        [MaxLength(500)]
        public string? Description { get; set; }

        [DataType(DataType.Date)]
        public DateTime StartDate { get; set; }

        [DataType(DataType.Date)]
        public DateTime? EndDate { get; set; }

        public bool IsCurrent { get; set; } = false;

        [MaxLength(100)]
        public string? Location { get; set; }

        public Student? Student { get; set; }
    }
}

File: student-portal/lib/model/experience.dart
class Experience {
  final int experienceId;
  final int studentId;
  final String companyName;
  final String role;
  final String? description;
  final DateTime startDate;
  final DateTime? endDate;
  final bool isCurrent;
  final String? location;

  Experience({
    required this.experienceId,
    required this.studentId,
    required this.companyName,
    required this.role,
    this.description,
    required this.startDate,
    this.endDate,
    required this.isCurrent,
    this.location,
  });

  factory Experience.fromJson(Map<String, dynamic> json) {
    return Experience(
      experienceId: json['experienceId'] ?? 0,

      // 🔹 FIX: Default to 0 to prevent "null is not subtype of int" crash
      studentId: json['studentId'] ?? 0,

      companyName: json['companyName'] ?? '',
      role: json['role'] ?? '',
      description: json['description'],

      // Safer Date Parsing
      startDate: json['startDate'] != null
          ? DateTime.tryParse(json['startDate'].toString()) ?? DateTime.now()
          : DateTime.now(),

      endDate: json['endDate'] != null
          ? DateTime.tryParse(json['endDate'].toString())
          : null,

      isCurrent: json['isCurrent'] ?? false,
      location: json['location'],
    );
  }

  // 🔹 ADDED: copyWith for Provider updates
  Experience copyWith({
    int? experienceId,
    int? studentId,
    String? companyName,
    String? role,
    String? description,
    DateTime? startDate,
    DateTime? endDate,
    bool? isCurrent,
    String? location,
  }) {
    return Experience(
      experienceId: experienceId ?? this.experienceId,
      studentId: studentId ?? this.studentId,
      companyName: companyName ?? this.companyName,
      role: role ?? this.role,
      description: description ?? this.description,
      startDate: startDate ?? this.startDate,
      endDate: endDate ?? this.endDate,
      isCurrent: isCurrent ?? this.isCurrent,
      location: location ?? this.location,
    );
  }
}
